waitress/channel.py

```diff
@@ -385,6 +414,15 @@ def service(self):
                         request.close()
                     self.requests = []
                 else:
+                    # before processing a new request, ensure there is not too
+                    # much data in the outbufs waiting to be flushed
+                    # XXX: currently readable() returns False while we are
```

Comment:
Don't use XXX, use NB please. XXX is something to fix, and this is basically just a notice to other developers.

Reply:
There's other code in this file that uses XXX that I think would fall under this argument but ok, I will change it.

Reply:
https://github.com/Pylons/waitress/blob/7f22411f24463eb170e256de4beccc644e712552/waitress/channel.py#L347

This is the other one in that file, and it's something that we should likely fix in due time, by tickling whatever wasyncore method we need to tickle to get it to attempt to write sooner.